src/Knp/Rad/Security/EventListener/AuthorizationListener.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Knp\Rad\Security\EventListener;
+
+use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+
+class AuthorizationListener
+{
+    /**
+     * @var AuthorizationCheckerInterface
+     */
+    private $checker;
+
+    /**
+     * @param AuthorizationCheckerInterface $checker
+     */
+    public function __construct(AuthorizationCheckerInterface $checker)
+    {
+        $this->checker = $checker;
+    }
+
+    public function checkIfUserIsGranted(FilterControllerEvent $event)
+    {
+        $request = $event->getRequest();
+
+        foreach ($request->attributes->get('_security', array()) AS $rule) {
+            $roles = array();
+
+            if (isset($rule['roles']) && !empty($rule['roles'])) {
+                $roles = $rule['roles'];
+            } else {
+                throw new \RuntimeException('You should provide "roles" parameter.');
+            }
+
+            if (is_string($roles)) {
+                $roles = array($roles);
+            }
+
+            $object = isset($rule['object']) ? $rule['object'] : null;
+
+            if ('&' === substr($object, 0, 1)) {
+                $objectName = substr($object, 1);
+
+                if (!$request->attributes->has($objectName)) {
+                    throw new \RuntimeException(sprintf("Object '%s' is not available.", $objectName));
+                }
+
+                $object = $request->attributes->get($objectName);
+            }
+
+            if (!$this->checker->isGranted($roles, $object)) {
+                throw new AccessDeniedHttpException();
```

Comment:
shouldn't you throw AccessDeniedException, which would be transformed in an http exception later